go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -376,6 +377,71 @@ type managementPortEntry struct {
 	config *managementPortConfig
 }
 
+// getEnvNameFromResourceName gets the device plugin env variable from the device plugin resource name.
+func getEnvNameFromResourceName(resource string) string {
+	res1 := strings.ReplaceAll(resource, ".", "_")
+	res2 := strings.ReplaceAll(res1, "/", "_")
+	return "PCIDEVICE_" + strings.ToUpper(res2)
+}
+
+// getDeviceIdsFromEnv gets the list of device IDs from the device plugin env variable.
+func getDeviceIdsFromEnv(envName string) ([]string, error) {
+	envVar := os.Getenv(envName)
+	if len(envVar) == 0 {
+		return nil, fmt.Errorf("unexpected empty env variable: %s", envName)
+	}
+	deviceIds := strings.Split(envVar, ",")
+	return deviceIds, nil
+}
+
+// handleDevicePluginResources tries to retrieve any device plugin resources passed in via arguments and device plugin env variables.
+func handleDevicePluginResources() error {
+	mgmtPortEnvName := getEnvNameFromResourceName(config.OvnKubeNode.MgmtPortDPResourceName)
+	deviceIds, err := getDeviceIdsFromEnv(mgmtPortEnvName)
+	if err != nil {
+		return err
+	}
+	// The reason why we want to store the Device Ids in a map is prepare for various features that
+	// require network resources such as the Management Port or Bypass Port. It is likely that these
+	// features share the same device pool.
+	config.OvnKubeNode.DPResourceDeviceIdsMap = make(map[string][]string)
+	config.OvnKubeNode.DPResourceDeviceIdsMap[config.OvnKubeNode.MgmtPortDPResourceName] = deviceIds
+	klog.V(5).Infof("Setting DPResourceDeviceIdsMap for %s using env %s with device IDs %v",
+		config.OvnKubeNode.MgmtPortDPResourceName, mgmtPortEnvName, deviceIds)
+	return nil
+}
+
+// takeDevicePluginInterface takes a device ID from a list of device IDs
+func takeDeviceId(deviceIds []string) (string, []string) {
+	ifsLen := len(deviceIds)
+	ret := (deviceIds)[ifsLen-1]
+	(deviceIds) = (deviceIds)[:ifsLen-1]
+	return ret, deviceIds
+}
+
+// handleNetdevResources tries to retrieve any device plugin interfaces to be used by the system such as the management port.
+func handleNetdevResources() error {
```

Comment:
> If we want to use this function for bypass port as well, need to move mgmt.port specific code. Let this function just return netdev name without checking/setting `MgmtPortNetdev` config, while actual config check done at line 564. 
Looks good I have implemented your suggested changes. Thanks!
> Notice, that resource name config doesn't work for DPU, because resource pool cannot be made from representors. For DPU-host it works, but DPU pod mgmt port netdev name must be manually fixed each time DPU-host restarted, because mgmt port device might have changed.
Yes, we are aware that resource name config doesn't work for DPU. The dpu-network-operator will be responsible for setting the management interface.

Reply:
Done.